robot-server/robot_server/runs/run_store.py

```diff
@@ -32,26 +35,58 @@ def __init__(self, run_id: str) -> None:
 class RunStore:
     """Methods for storing and retrieving run resources."""
 
-    def __init__(self) -> None:
-        """Initialize a RunStore and its in-memory storage."""
-        self._runs_by_id: Dict[str, RunResource] = {}
+    def __init__(self, sql_engine: sqlalchemy.engine.Engine) -> None:
+        """Initialize a RunStore with sql engine."""
+        self._sql_engine = sql_engine
+        self._active_run = ""
 
-    def upsert(self, run: RunResource) -> RunResource:
-        """Insert or update a run resource in the store.
+    def insert_action(self, run_id: str, action: RunAction) -> None:
+        """Insert run action in the db.
 
         Arguments:
-            run: Run resource to store. Reads `run.id` to
-                determine identity in storage.
+            run_id: current run id to get
+            action: action to insert into the db
+        """
+        with self._sql_engine.begin() as transaction:
+            _insert_action_no_transaction(run_id, action, transaction)
+
+    def _get_actions(self, run_id: str) -> List[RunAction]:
+        with self._sql_engine.begin() as statement:
+            actions = _get_actions_no_transaction(run_id, statement)
+        return [_convert_sql_row_to_action(action) for action in actions]
+
+    def update_active_run(self, run_id: str, is_current: bool) -> RunResource:
+        """Update current active run resource in memory.
+
+        Arguments:
+            run_id: run to update
+            is_current: is run active or not
+
+        Returns:
+            The resource that was updated.
+        """
+        if is_current is True:
+            self._active_run = run_id
+        elif is_current is False and self._active_run == run_id:
+            self._active_run = ""
```

Comment:
```suggestion
            self._active_run = None
```

(Matching my other comment about using `None` when no run is current.)